packages/zwave-js/src/lib/commandclass/PowerlevelCC.ts

```diff
@@ -0,0 +1,409 @@
+			this.timeout = this.payload[1];
+		}
+
+		this.persistValues();
+	}
+
+	@ccValue({ stateful: false })
+	@ccValueMetadata({
+		...ValueMetadata.UInt8,
+		states: enumValuesToMetadataStates(Powerlevel),
+		label: "Power level",
+		description: "The current power level  in effect on the node.",
+	})
+	public readonly powerlevel: Powerlevel;
+
+	@ccValue({ stateful: false })
+	@ccValueMetadata({
+		...ValueMetadata.UInt8,
+		min: 1,
+		max: 255,
+		label: "Timeout",
+		description: "Time in seconds before node returns to Normal power.",
+	})
+	public readonly timeout?: number;
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: { "Power level": this.powerlevel, Timeout: this.timeout },
+		};
+	}
+}
+
+@CCCommand(PowerlevelCommand.Get)
+@expectedCCResponse(PowerlevelCCReport)
+export class PowerlevelCCGet extends PowerlevelCC {}
+
+interface PowerlevelCCTestNodeSetOptions extends CCCommandOptions {
+	testNodeId: number;
+	powerlevel: Powerlevel;
+	testFrameCount: number;
+}
+
+@CCCommand(PowerlevelCommand.TestNodeSet)
+export class PowerlevelCCTestNodeSet extends PowerlevelCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| PowerlevelCCTestNodeSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			const testNode = driver.controller.nodes.get(options.testNodeId);
+			if (!testNode) {
+				throw new ZWaveError(
+					`${this.constructor.name}: Node Id ${options.testNodeId} doesn't exist.`,
+					ZWaveErrorCodes.Argument_Invalid,
+				);
+			}
+			if (testNode.isFrequentListening) {
+				throw new ZWaveError(
+					`${this.constructor.name}: Node Id ${options.testNodeId} is FLiRS, which is not allowed for the Test Node.`,
+					ZWaveErrorCodes.Argument_Invalid,
+				);
+			}
+			if (testNode.status !== NodeStatus.Awake) {
+				throw new ZWaveError(
+					`${this.constructor.name}: Node Id ${options.testNodeId} status is ${testNode.status}, but the Test Node must be Awake.`,
+					ZWaveErrorCodes.Argument_Invalid,
+				);
+			}
+
+			this.testNodeId = options.testNodeId;
+			this.powerlevel = options.powerlevel;
+			this.testFrameCount = options.testFrameCount;
+		}
+	}
+
+	public testNodeId: number;
+	public powerlevel: Powerlevel;
+	public testFrameCount: number;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.from([
+			this.testNodeId,
+			this.powerlevel,
+			(this.testFrameCount >> 8) & 0xff,
+			this.testFrameCount & 0xff,
+		]);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: {
+				testNodeId: this.testNodeId,
+				powerlevel: this.powerlevel,
+				testFrameCount: this.testFrameCount,
```

Comment:
the property names should be a bit nicer in a log: `test node ID`, `powerlevel`, `test frame count`.